src/cascadia/TerminalSettingsEditor/Profiles.xaml

```diff
@@ -79,16 +79,18 @@ the MIT License. See LICENSE in the project root for license information. -->
                                     <TextBox x:Uid="Profile_StartingDirectory"
                                          x:Name="StartingDirectory"
                                          Text="{x:Bind State.Profile.StartingDirectory, Mode=TwoWay}"
-                                         Style="{StaticResource TextBoxSettingStyle}"/>
+                                         Style="{StaticResource TextBoxSettingStyle}"
+                                         IsEnabled="{x:Bind State.Profile.UseCustomStartingDirectory, Mode=OneWay}"/>
                                     <Button x:Uid="Profile_StartingDirectoryBrowse"
                                         x:Name="StartingDirectoryBrowse"
                                         Click="StartingDirectory_Click"
+                                        IsEnabled="{x:Bind State.Profile.UseCustomStartingDirectory, Mode=OneWay}"

```

Comment:
Should we keep the button enabled? If you browse for and choose a directory, it will set the StartingDirectory and disable the checkbox, right?

Reply:
This way we get consistency with the way background image path is done though!

Also it turns out what happens is if you click browse while the checkbox is checked and choose a directory, the textbox ends up getting set to `%USERPROFILE%` (or the cached starting directory path if there was one) - this does not happen if you browse when the checkbox is not set however.

Reply:
Is that not because you're setting Checked(false) and triggering a cascading two-way binding that calls into "UseParentStartingDir(false)"? They're effectively racing.

Reply:
Its not - I had already removed that line (line 347) when I tested this

Reply:
The reason this happens is that when the checkbox is not checked and we browse->select a folder, we don't even get into UseParentProcessDirectory; but if the checkbox is checked and we browse->select a folder, then we do get into UseParentProcessDirectory which then resets the path